src/Cli/Microsoft.TemplateEngine.Cli/TemplatePackageCoordinator.cs

```diff
@@ -556,5 +805,371 @@ private async Task<bool> IsTemplateShortNameAsync(string sourceIdentifier, Cance
+                {
+                    Reporter.Output.WriteLine(constraintResult.ToDisplayString().Indent(1));
+                }
+            }
+        }
+
+        private void HandleUpdateCheckErrors(CheckUpdateResult result, bool ignoreLocalPackageNotFound)
+        {
+            switch (result.Error)
+            {
+                case InstallerErrorCode.InvalidSource:
+                    Reporter.Error.WriteLine(
+                        string.Format(
+                            LocalizableStrings.TemplatePackageCoordinator_Update_Error_InvalidNuGetFeeds,
+                            result.TemplatePackage.DisplayName).Bold().Red());
+                    break;
+                case InstallerErrorCode.PackageNotFound:
+                    if (!ignoreLocalPackageNotFound || !result.TemplatePackage.IsLocalPackage)
+                    {
+                        Reporter.Error.WriteLine(
+                            string.Format(
+                                LocalizableStrings.TemplatePackageCoordinator_Update_Error_PackageNotFound,
+                                result.TemplatePackage.DisplayName).Bold().Red());
+                    }
+                    break;
+                case InstallerErrorCode.UnsupportedRequest:
+                    Reporter.Error.WriteLine(
+                        string.Format(
+                            LocalizableStrings.TemplatePackageCoordinator_Update_Error_PackageNotSupported,
+                            result.TemplatePackage.DisplayName).Bold().Red());
+                    break;
+                case InstallerErrorCode.GenericError:
+                default:
+                    Reporter.Error.WriteLine(
+                        string.Format(
+                            LocalizableStrings.TemplatePackageCoordinator_Update_Error_GenericError,
+                            result.TemplatePackage.DisplayName,
+                            result.ErrorMessage).Bold().Red());
+                    break;
+            }
+        }
+
+        private IEnumerable<PackageSource> LoadNuGetSources(IEnumerable<string>? additionalSources, bool includeNuGetFeed)
+        {
+            IEnumerable<PackageSource> defaultSources;
+            string currentDirectory = string.Empty;
+            try
+            {
+                currentDirectory = Directory.GetCurrentDirectory();
+                ISettings settings = global::NuGet.Configuration.Settings.LoadDefaultSettings(currentDirectory);
+                PackageSourceProvider packageSourceProvider = new PackageSourceProvider(settings);
+                defaultSources = packageSourceProvider.LoadPackageSources().Where(source => source.IsEnabled);
+                if (includeNuGetFeed)
+                {
+                    var nuGetFeed = new PackageSource(NugetOrgFeed);
+                    defaultSources = defaultSources.Append(nuGetFeed);
+                }
+            }
```

Comment:
Future expansion: I think if `string packageIdentity` was added to this method, then we could pretty easily add support for PackageSourceMapping to this feature via something like this:

```csharp
PackageSourceMapping packageSourceMapping = PackageSourceMapping.GetPackageSourceMapping(settings);
IReadOnlyList<string> validPackageSources = packageSourceMapping.GetConfiguredPackageSources(packageIdentity);
defaultSources = packageSourceProvider.LoadPackageSources().Where(source => source.IsEnabled && (validPackageSources?.Contains(source.Name) ?? true));
```

We should consider PSM support all-up for templating, though.